### PR TITLE
[docs] Using MultiClusterIngress API

### LINKED ISSUE #1079: Support Multi-Cluster Ingress
**Please provide an in-depth description of the question you have**:
   K8s has signal-cluster ingress controller that forward traffic to services. 
   Does Karmada have mulit-cluster counterparts or not?
**What do you think about this question?**:
   1.If necessary, how to integrate with karmada.
**Environment**:
- Karmada version: v0.10.1
- Kubernetes version: v1.22.3
- Others:

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Add docs for using MultiClusterIngress API to import external traffic to services in the member clusters.

**Which issue(s) this PR fixes**:
Fixes #1079 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

/cc @RainbowMango @GitHubxsy @kevin-wangzefeng

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @Tingtal, can you help review it, thanks~

File: docs/multi-cluster-ingress.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+So we need to ensure that the container networks between the **host cluster** and member clusters are connected.  The **host cluster** indicates the cluster where the Karmada control plane is deployed.
+
+- If you use the `hack/local-up-karmada.sh` script to deploy Karmada, Karmada will have three member clusters, and the container networks between the **host cluster**, `member1` and `member2` are connected.
+- You can use `Submariner` or other related open source projects to connected networks between clusters.
+
+## Example
+
+### Step 1: Deploy ingress-nginx on the host cluster
+
+We use [ingress-nginx](https://github.com/kubernetes/ingress-nginx) as the demo for demonstration. We've made some changes based on the latest version(controller-v1.1.1) of ingress-nginx, for detail modification you can see [here](https://github.com/karmada-io/multi-cluster-ingress-nginx/tree/main).
```
Comment (RainbowMango) [inline docs/multi-cluster-ingress.md:24]: You are using `multi cluster ingress nginx` now not `ingress-nginx`.

File: docs/multi-cluster-ingress.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+kubectl config use-context karmada-host
+make dev-env
+```
+
+#### Apply kubeconfig secret
+
+Create a secret that contains the `karmada-apiserver` authentication credential in the following format:
+
+```yaml
+# karmada-kubeconfig.yaml
```
Comment (RainbowMango) [inline docs/multi-cluster-ingress.md:50]: `karmada-kubeconfig.yaml`??
Looks like it's a `KUBECONFIG`, not a `Secret`.

File: docs/multi-cluster-ingress.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+kubectl -n karmada-system get secret kubeconfig -oyaml | grep kubeconfig | sed -n '1p' | awk '{print $2}'
+```
+
+Then apply yaml:
+
+```
+kubectl apply -f karmada-kubeconfig.yaml
+```
+
+#### Edit ingress-nginx-controller deployment
```
Comment (RainbowMango) [inline docs/multi-cluster-ingress.md:73]: Why do people need to edit it manually? Add some description here.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

File: docs/multi-cluster-ingress.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+```
+// for HTTPS
+git clone https://github.com/karmada-io/multi-cluster-ingress-nginx.git
+// for SSH
+git clone git@github.com:karmada-io/multi-cluster-ingress-nginx.git
+```
+
+#### Build and deploy ingress-nginx
+
+Using existing `karmada-host` kind cluster to building and deploying the ingress controller.
```
Comment (Tingtal) [inline docs/multi-cluster-ingress.md:37]: Use the existing `karmada-host` kind cluster to build and deploy the ingress controller.

File: docs/multi-cluster-ingress.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+#### Apply kubeconfig secret
+
+Create a secret that contains the `karmada-apiserver` authentication credential in the following format:
+
+```yaml
+# karmada-kubeconfig-secret.yaml
+apiVersion: v1
+data:
+  kubeconfig: {data} # encoded by base64
```
Comment (Tingtal) [inline docs/multi-cluster-ingress.md:53]: kubeconfig: {data} # Base64-encoded

File: docs/multi-cluster-ingress.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+### Karmada has been installed
+
+We can install Karmada by referring to [quick-start](https://github.com/karmada-io/karmada#quick-start), or directly run `hack/local-up-karmada.sh` script which is also used to run our E2E cases.
+
+### Cluster Network
+
+Currently, we need to use the [MCS](https://github.com/karmada-io/karmada/blob/master/docs/multi-cluster-service.md#the-serviceexport-and-serviceimport-crds-have-been-installed) feature to import external traffic.
+
+So we need to ensure that the container networks between the **host cluster** and member clusters are connected.  The **host cluster** indicates the cluster where the Karmada control plane is deployed.
```
Comment (Tingtal) [inline docs/multi-cluster-ingress.md:15]: So we need to ensure that the container networks between the **host cluster** and member clusters are connected. The **host cluster** indicates the cluster where the Karmada control plane is deployed.

File: docs/multi-cluster-ingress.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+
+Then apply yaml:
+
+```
+kubectl apply -f karmada-kubeconfig-secret.yaml
+```
+
+#### Edit ingress-nginx-controller deployment
+
+We want `nginx-ingress-controller` to access `karmada-apiserver` to listen to resource(such as multiclusteringress, endpointslices, service, etc.) changes, therefore, we need to mount the authentication credential of `karmada-apiserver` to the `nginx-ingress-controller`.
```
Comment (Tingtal) [inline docs/multi-cluster-ingress.md:75]: We want `nginx-ingress-controller` to access `karmada-apiserver` to listen to changes in resources (such as multiclusteringresses, endpointslices, and services). Therefore, we need to mount the authentication credential of `karmada-apiserver` to the `nginx-ingress-controller`.

File: docs/multi-cluster-ingress.md [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+          name: kubeconfig                      # new line
+          subPath: kubeconfig                   # new line
+      volumes:
+	  ...
+      - name: kubeconfig                        # new line
+        secret:                                 # new line
+          secretName: kubeconfig                # new line
+```
+
+### Step 2: Use MCS feature to discovery service
```
Comment (Tingtal) [inline docs/multi-cluster-ingress.md:109]: ### Step 2: Use the MCS feature to discover services

**Review by Tingtal — COMMENTED** [review]
Grammatical fixes provided.

**Pull request merged**